PYME/LMVis/Extras/animation.py

```diff
@@ -198,18 +242,27 @@ def play(self, save):
                         new_view = current_view + difference_view * step
                         self.get_canvas().set_view(new_view.normalize_view())
                         if save:
-                            snap = self.get_canvas().getIm()#.astype('uint8')
+                            snap = self.get_canvas().getIm() #pixel_size=float(self.pixel_size.GetValue()))#.astype('uint8')
                             #print snap.shape, snap.dtype, snap.min(), snap.max()
-                            snap = (255*snap).astype('uint8').transpose(1, 0, 2)
+                            if snap.dtype == 'uint8':
+                                # PYME.LMVis.gl_render3D_shaders already returns 
+                                # a snapshot as an 8-bit array
+                                snap = snap.transpose(1,0,2)
+                            else:
+                                snap = (255*snap).astype('uint8').transpose(1, 0, 2)
                             
                             #if snap.ndim == 3:
                             #    video.write(cv2.cvtColor(cv2.flip(snap.transpose(1, 0, 2), 0), cv2.COLOR_RGB2BGR))
                             #else:
                             #    video.write(cv2.flip(snap.transpose(1, 0, 2), 0))
-                            im = Image.fromarray(snap).save(os.path.join(dir_name, 'frame%04d.jpg' % f_no))
+                            dpi = int(float(self.dpi.GetValue()))

```

Comment:
Is dpi a) useful? and b) doing what you expect? 

When passed to PIL it simply writes the value to the image metadata - it doesn't rescale the image. DPI is only really useful when printing the images (and then arguably only if it's important if the physical size of the images is important - i.e. to make sure that 12pt text is the right size etc ...). I can't really see dpi being at all relevant for video production or the majority of scientific workflows where you arrange images into subpanels with illustrator etc ... The only real time you need to worry about DPI in a scientific workflow is wen exporting image composites to figures in for papers - but that then has very little relation to the input dpi attributed to each of the panels.

You've also got the various confusions with retina displays etc (where the default ppi is 144 rather than 72), so my gut feeling is that you are adding an extra field which in practice does very little, but might not be populated with correct default values if you do for some reason want to record the physical size of the image on screen. For dpi to be at all meaningful, you'd need to interrogate the display to see what it's actual ppi is (which will be super hard to do in a platform independant way) and then re-scale the image to the desired output dpi. I'm not sure why you'd want to do this.

Reply:
I assumed the DPI would actually change the DPI of the image for figure rendering. Since it's only metadata, this is probably useless. Thanks for pointing this out.

I'd like to be able to change the snapshot pixel size or scaling (which I'd also change by just modifying `pixel_size`). As it stands, the `pixel_size` argument in `PYME.LMVis.gl_render3D_shaders.LMGLShaderCanvas.getIm()` does not scale the image in a useful way, but instead crops the image. I propose a method of adjusting this in #283. Please let me know if you think this is the right approach.

Reply:
What is the rationale for needing to change the pixel size / scaling in animation? Surely the animation is just a visual and it's unimportant what the actual pixel size is - the only reason for changing sizes is then to optimize the size on disc / fit in a powerpoint slide etc ... The easiest approach here is going to be to re-size the VisGUI window to the desired size  before capturing the movie (this has the added advantage of letting you adjust the aspect ratio at the same time).

A physical calibration of pixel size and distances is not trivial for 3D orthographic views, and not possible for 3D perspective (which we plan to make more visible in the GUI), and as a result I think that any implication in the UI that you can generate an animation scaled in anything other than pixel units is not desirable (we can have scale bars or, for 3D, scale boxes in the image).

For most animation use cases, the best way of displaying zoom settings would potentially be to offer a dropdown of preset output sizes in pixels - e.g. current (`glCanvas.Size[0]`, `glCanvas.Size[1]`), 640x480, 320x240, etc, potentially including a custom option where you could set the number of pixels in x and y. So the user experience is that you get what you see on screen, but at the pixel sizing you desire. A caveat of this is that you are going to need to deal with the aspect ratio potentially being different (I'd opt for the conservative approach where you include all of the current viewport and potentially a bit extra, rather than cropping down to the new aspect ratio), so animation no longer becomes WYSIWYG. A lot of this could alternatively be handled by documentation saying to resize to the window before capturing though. The only one case I can see that resizing the window first doesn't address is if you want the animation at full screen resolution.

Reply:
We had discussed exposing `pixel_size` or dpi in #224. I think a `scaling` option is pretty standard for snapshot exports (e.g. Meshlab). That said, we don't need to have it here. 

What do you think about accepting this pull request (or make notes on what else needs to be changed) and we'll revisit how to handle scaling in a different PR?

Reply:
sounds like a good plan. I'm not adverse to having an option for scaling of the output, just want it to make sense if and when it goes in.